### ISSUE TITLE
Does not work on pages handled by a mounted engine

### ISSUE THREAD

**Issue by mschuerig** [MiniProfiler/rack-mini-profiler#25]
For pages handled by a mounted engine, the inserted script element looks like this

    <script async="" type="text/javascript" id="mini-profiler" src="/foo-engine/mini-profiler-resources/includes.js?v=..." ...></script>

Note the source attribute, it is prefixed by "/foo-engine" (the mythical example engine). Unfortunately, rmp does not provide its script at that locating. The script is supposed to be at /mini-profiler-resources/includes.js.

As far as I can tell, the reason for the wrong path is that Rails sets env['SCRIPT_NAME'] to point at the engine's mount point. That parameter in turn is used by rmp to set up its script path.

A very hackish woraround I'm currently using is this change in Rack::MiniProfiler#call

      #return serve_html(env) if path.start_with? @config.base_url_path
      return serve_html(env) if path.include? @config.base_url_path

**Comment by albertodega** [comment]
+1

**Comment by th3sku11** [comment]
+1

**Comment by nspring** [comment]
The most recent change (c125062) appears to break running under passenger with RackBaseURI set.  I don't entirely understand the reasoning behind the conditional, but it evaluates to true, even though url_for does not then add the necessary SCRIPT_NAME prefix to allow the assets to be loaded.  I believe this is also experienced by others as in [another issue described here](http://community.miniprofiler.com/t/passenger-rackbaseuri-issues-with-paths/419).

Does that commit seem like a really good idea?  Does 2b1484a not solve the problem on its own?  I'd create a pull request to revert the change, but I assume it helps in some situation.

**Comment by nspring** [comment]
As a follow-up, it appears that url_for, when given a String argument necessarily returns just that string. Invoking it to "correctly construct the path" as the comment suggests seems like it will necessarily fail.
See [url_for source](https://github.com/rails/rails/blob/35842aa91883245dbf4270b0fc863684aabf4688/actionpack/lib/action_dispatch/routing/url_for.rb#L150)

**Comment by SamSaffron** [comment]
@nspring can you submit a patch for this? 

I am so lost on these issues, it appears the test matrix here is hellish, we need some proper tests for all the crazy situations here cc @mschuerig

**Comment by nspring** [comment]
I've uploaded what I think is a minimal way to address the problem of running under Passenger with a different base URI.  I don't want to inadvertently break the rails engine use case that solicited all the +1's above.  I think this is safe, except perhaps when Passenger and Rails Engines are used at the same time, though I suspect this will work correctly.

**Comment by SamSaffron** [comment]
@nspring yeah most recent change also broke Discourse under a subfolder, I got to get it sorted

**Comment by mtomov** [comment]
And to add to the drama, 

The expectation written:

```
mock_controller.should_receive(:url_for).with('/mini-profiler-resources/').and_return('/test/mini-profiler-resources/')
```

is indeed not valid as @nspring already pointed out. I have a Rails app mounted at a path, and `url_for('mini-profile-resources')` does not prepend `test` as expected. @nspring gave a link to the rails source code of `url_for`, but I'm not sure if the implementation of it might not even be just that really:

- http://apidock.com/rails/ActionController/Metal/url_for

Either way, reading up, the confusion comes from the desire to remove `SCRIPT_NAME` when using engines as pointed out by the author:

> Note the source attribute, it is prefixed by "/foo-engine" (the mythical example engine). Unfortunately, rmp does not provide its script at that locating. The script is supposed to be at /mini-profiler-resources/includes.js.

However, I think that removing `foo-engine` from the path is not the correct approach, and rather the problem lies in the profiler not providing its script at the mounted location.

In my app, requesting `/test/mini-profiler-resources` actually works just fine, but I can't tell if this is "default behaviour" or one of my app settings adjusts that ( `map '/test/" do;  run Rails.application; end` in `config.ru` for example )
